### ISSUE TITLE
Unable to install OpenZFS using zfsinstaller.exe

### ISSUE THREAD

**Issue by arun-kv** [openzfsonwindows/openzfs#2]
I was trying to install OpenZFS using zfsinstaller.exe, and getting below error.

PS Z:\out\install\x64-Debug\bin\driver> .\zfsinstaller.exe install .\OpenZFS.inf
[SC] EnumQueryServicesStatus:OpenService FAILED 1060:

The specified service does not exist as an installed service.

OpenZFS_Install 128 .\OpenZFS.inf
[SC] EnumQueryServicesStatus:OpenService FAILED 1060:

The specified service does not exist as an installed service.

StartServiceA failed, error 487
installRootDevice: exit 0:0x0
Installation done.

![image](https://user-images.githubusercontent.com/65647132/106722656-39d84100-662c-11eb-8357-ec3c1c339b0f.png)

**Comment by lundman** [comment]
OK, so lets see if we can get it to work. It does have some debugging, so the first "error" is expected, it is testing to see if the service is already running.

I'll see if I can get it run in my target VM today

**Comment by lundman** [comment]
Hmm it is not behaving like it used to.   I have it setup to run zfsisntaller.exe with
```
      "type": "default",
      "project": "CMakeLists.txt",
      "projectTarget": "zfsinstaller.exe (cmd\\os\\windows\\zfsinstaller\\zfsinstaller.exe)",
      "name": "zfsinstaller.exe (cmd\\os\\windows\\zfsinstaller\\zfsinstaller.exe)",
      "args": [
        "install",
        "..\\..\\..\\..\\module\\os\\windows\\driver\\OpenZFS.inf"
      ]
    }
```

Which will ask if I want to install the SYS file, and if my certificate is OK, copies the file to Windows...

The services fail, so output is:
```
[SC] EnumQueryServicesStatus:OpenService FAILED 1060:

The specified service does not exist as an installed service.



OpenZFS_Install 128 ..\..\..\..\module\os\windows\driver\OpenZFS.inf
[SC] EnumQueryServicesStatus:OpenService FAILED 1060:

The specified service does not exist as an installed service.

StartServiceA failed, error 487
installRootDevice: exit 0:0x0
Installation done.
```

But installRootDevice is good - however, it does not start the SYS file (debugger gets no information)

So probably need to solve why the services thing fail.

**Comment by lundman** [comment]
OK, so 487 or invalid address - is because it gets built wrong with CMake. If I copy just the openzfs.sys file (linked by the deploy project) it loads fine. Next up is figuring out what is missing in the CMake building.

**Comment by lundman** [comment]
Yeah the cmake thing compiles everything as clang, which is not right - it is supposed to do driver.c (plus three other files) and then link, using visual studio compiler.

**Comment by lundman** [comment]
Working on it, turns out that it has been compiling driver.c with clang, and worse for me, I've been compiling with clang-8, and not the clang-10 I should be using. Cleaning things up now

**Comment by lundman** [comment]
OK sorted, clang-10 also picks -m32 by default, so we need to set -m64. Possibly we will have to go back at check 32bit builds still build, but it's pretty low priority for me. Could also look at Release build at this point, but I'll be living in DEBUG for a while until it is stable.

With ff5e232 I can now build it on VM, then load it with zfsinstaller. Sorry about the delay getting that right.

I do still get this:
```
C:\src\openzfs\out\build\x64-Debug\EXEC : error : 
```
Which is possibly related to the gitrev build in cmake? But it doesn't actually stop it from building.

**Comment by imtiazdc** [comment]
@lundman Excellent progress. Thank you!

FYI, we were able to build x64-Clang-Release configuration the other day. If there are any changes we need to make to release configuration because of your recent changes to make zfsinstaller debug version work, then please let us know and we will fix it on our end.

**Comment by lundman** [comment]
It theory it should just be DBG vs NDEBUG defines I guess.. shouldn't be anything magical in there one hopes, I've just not tried it myself.
